spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -51,30 +51,56 @@
 
 	private ConcurrentHashMap<String, Publisher> publishers = new ConcurrentHashMap<>();
 
-	private ExecutorProvider executorProvider;
-
 	private MessageConverter messageConverter = new SimpleMessageConverter();
 
-	private int concurrentProducers = 1;
+	/**
+	 * The Spring component a Google Cloud Pub/Sub call is originated from.
+	 *
+	 * <p>Spring Integration callers should set this to "spring-integration", Spring Cloud to
+	 * "spring-cloud", etc.
+	 */
+	private String sourceName = "spring";
```

Comment:
Looks like it should be a constant - `SOURCE_NAME`, and the same for `SOURCE_VERSION`.